mirai-core-api/src/commonMain/kotlin/utils/DeviceInfo.kt

```diff
@@ -46,8 +46,40 @@ public expect class DeviceInfo(
     wifiSSID: ByteArray,
     imsiMd5: ByteArray,
     imei: String,
-    apn: ByteArray
+    apn: ByteArray,
+    androidId: ByteArray,
 ) {
+    @Deprecated(
+        "This DeviceInfo constructor may randomize field `androidId` without your random instance. " +
+                "It is better to specify `androidId` explicitly.",
+        replaceWith = ReplaceWith(
+            "net.mamoe.mirai.utils.DeviceInfo(display, product, device, board, brand, model, " +
+                    "bootloader, fingerprint, bootId, procVersion, baseBand, version, simInfo, osType, " +
+                    "macAddress, wifiBSSID, wifiSSID, imsiMd5, imei, apn, )"
+        )
+    )
```

Comment:
```suggestion
    @Deprecated(
        "This DeviceInfo constructor may randomize field `androidId` without your random instance. " +
                "It is better to specify `androidId` explicitly.",
        replaceWith = ReplaceWith(
            "net.mamoe.mirai.utils.DeviceInfo(display, product, device, board, brand, model, " +
                    "bootloader, fingerprint, bootId, procVersion, baseBand, version, simInfo, osType, " +
                    "macAddress, wifiBSSID, wifiSSID, imsiMd5, imei, apn, androidId)"
        )
    )
```

你后面得放个东西, 不然 replace 之后的调用还是这个旧构造器